react-effects/my-app/src/List.jsx

```diff
@@ -0,0 +1,41 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import readItems from './read';
+
+export default function List() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [items, setItems] = useState([]);
+  const [error, setError] = useState();
+
+  useEffect(()=>{
+    async function startFetching() {
+      try {
+        const result = await readItems();
+        setItems(result);
+        setIsLoading(false)
+      } catch (error) {
+        setError(error);
+        setIsLoading(false);
```

Comment:
Instead of having to write `setIsLoading(false)` twice, we can use a finally block! 

```
async function startFetching() {
      try {
        const result = await readItems();
        setItems(result);
      } catch (error) {
        setError(error);
      } finally {
        setIsLoading(false);
      }
    }
```